src/thingset.h

```diff
@@ -71,74 +71,98 @@ extern "C" {
  *     0 .. 3: High-priority control frames
- *     4 .. 7: Normal pub/sub frames for monitoring
+ *     5, 7: Normal pub/sub frames for monitoring
+ *
+ * Network management (e.g. address claiming):
+ *
+ *    28      26 25 24 23           16 15            8 7             0
+ *   +----------+-----+---------------+---------------+---------------+
+ *   | Priority | 0x3 |  random data  |  target addr  |  source addr  |
+ *   +----------+-----+---------------+---------------+---------------+
+ *
+ *   Priority: 4
  */
 
-#define TS_CAN_SOURCE_POS               (0U)
-#define TS_CAN_SOURCE_MASK              (0xFF << TS_CAN_SOURCE_POS)
-#define TS_CAN_SOURCE_SET(addr)         ((uint32_t)addr << TS_CAN_SOURCE_POS)
-
-#define TS_CAN_TARGET_POS               (8U)
-#define TS_CAN_TARGET_MASK              (0xFF << TS_CAN_TARGET_POS)
-#define TS_CAN_TARGET_SET(addr)         ((uint32_t)addr << TS_CAN_TARGET_POS)
-
-#define TS_CAN_DATA_ID_POS              (8U)
-#define TS_CAN_DATA_ID_MASK             (0xFFFF << TS_CAN_DATA_ID_POS)
-#define TS_CAN_DATA_ID_SET(id)          ((uint32_t)id << TS_CAN_DATA_ID_POS)
-
-#define TS_CAN_J1939_DP                 (0x1 << 24U)
-#define TS_CAN_J1939_EDP                (0x1 << 25U)
-
-#define TS_CAN_TYPE_REQRESP             (TS_CAN_J1939_EDP)
-#define TS_CAN_TYPE_PUBSUB              (TS_CAN_J1939_EDP | TS_CAN_J1939_DP)
-#define TS_CAN_TYPE_MASK                (TS_CAN_J1939_EDP | TS_CAN_J1939_DP)
-
+/* source and target addresses */
+#define TS_CAN_SOURCE_POS           (0U)
+#define TS_CAN_SOURCE_MASK          (0xFF << TS_CAN_SOURCE_POS)
+#define TS_CAN_SOURCE_SET(addr)     (((uint32_t)addr << TS_CAN_SOURCE_POS) & TS_CAN_SOURCE_MASK)
+#define TS_CAN_SOURCE_GET(id)       (((uint32_t)id & TS_CAN_SOURCE_MASK) >> TS_CAN_SOURCE_POS)
+
+#define TS_CAN_TARGET_POS           (8U)
+#define TS_CAN_TARGET_MASK          (0xFF << TS_CAN_TARGET_POS)
+#define TS_CAN_TARGET_SET(addr)     (((uint32_t)addr << TS_CAN_TARGET_POS) & TS_CAN_TARGET_MASK)
+#define TS_CAN_TARGET_GET(id)       (((uint32_t)id & TS_CAN_TARGET_MASK) >> TS_CAN_TARGET_POS)
+
+#define TS_CAN_ADDR_MAX             (0xFD)
+#define TS_CAN_ADDR_ANONYMOUS       (0xFE)
+#define TS_CAN_ADDR_BROADCAST       (0xFF)
+
+/* data IDs for publication messages */
+#define TS_CAN_DATA_ID_POS          (8U)
+#define TS_CAN_DATA_ID_MASK         (0xFFFF << TS_CAN_DATA_ID_POS)
+#define TS_CAN_DATA_ID_SET(id)      (((uint32_t)id << TS_CAN_DATA_ID_POS) & TS_CAN_DATA_ID_MASK)
+#define TS_CAN_DATA_ID_GET(id)      (((uint32_t)id & TS_CAN_DATA_ID_MASK) >> TS_CAN_DATA_ID_POS)
+
+/* bus number for request/response messages */
+#define TS_CAN_BUS_NO_POS           (16U)
+#define TS_CAN_BUS_NO_MASK          (0xFF << TS_CAN_BUS_NO_POS)
+#define TS_CAN_BUS_NO_SET(id)       (((uint32_t)id << TS_CAN_BUS_NO_POS) & TS_CAN_BUS_NO_MASK)
+#define TS_CAN_BUS_NO_GET(id)       (((uint32_t)id & TS_CAN_BUS_NO_MASK) >> TS_CAN_BUS_NO_POS)
+#define TS_CAN_BUS_NO_DEFAULT       (218U) // 0xDA, N_TAtype = physical
```

Comment:
It might be clearer to name these BUS_NUM instead of BUS_NO - Taken out of context for example TS_CAN_BUS_NO_DEFAULT looks like something without a default. But this is incredibly minor :)

Reply:
Thanks for the review. That's a good point. I called it `BUS_ID` now, so it is consistent with `DATA_OBJECT_ID`.